tensorhive/controllers/schedule/update_schedule_controller.py

```diff
@@ -0,0 +1,49 @@
+from datetime import datetime
+from tensorhive.utils.Weekday import Weekday
+from tensorhive.authorization import admin_required
+from tensorhive.models.RestrictionSchedule import RestrictionSchedule
+from sqlalchemy.orm.exc import NoResultFound
+from tensorhive.config import API
+import logging
+log = logging.getLogger(__name__)
+S = API.RESPONSES['schedule']
+G = API.RESPONSES['general']
+
+
+@admin_required
+def update(id, newValues):
+    new_values = newValues
+    allowed_fields = {'scheduleDays', 'hourStart', 'hourEnd'}
+    try:
+        assert set(new_values.keys()).issubset(allowed_fields), 'invalid field is present'
+        schedule = RestrictionSchedule.get(id)
+
+        for field_name, new_value in new_values.items():
+            # Mapping API field name to column used by Schedule model
+            if field_name == 'scheduleDays':
+                field_name = 'schedule_days'
+                days = []
+                for day in new_value:
+                    days.append(Weekday[day])
+                new_value = days
+            if field_name in ['hourStart', 'hourEnd']:
+                # hourStart -> hour_start, hourEnd -> hour_end
+                field_name = 'hour_' + (field_name[4:]).lower()
```

Comment:
See my previous comment in update restriction controller... In its current form I could probably bet some money on someone forgetting to update this part of code if he/she ever wanted to change name of these columns in the DB model